### ISSUE TITLE
Restore files' original last modified time?

### ISSUE THREAD

**Issue by LuLu010722** [actions/checkout#968]
Every time someone clones other's repo, the last modified time of each file and directory is overwritten with the time of clone.
**However**, in some cases, we need to know the original time infomation, such as in a blog static site.

There are ways to fix this.
For example, run `git ls-files -z | while read -d '' path; do touch -d $(git log -1 --format="@%ct" "$path") "$path"; done` in the workflow file.
But, can you integrate this feature inside actions/checkout and provide a config option like "origin-mtime"?

**Comment by yu-iskw** [comment]
+1

We can also utilize a community-based Action as https://github.com/marketplace/actions/git-restore-mtime . However, it would be great for the `checkout` Action officially supports.

**Comment by theory** [comment]
Yes please, this breaks the ability for modified time-based build systems like `make` to use files restored from actions/cache.
